qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -25,17 +30,40 @@ class B1517A(B1500Module):
     def __init__(self, parent: 'KeysightB1500', name: Optional[str], slot_nr,
                  **kwargs):
         super().__init__(parent, name, slot_nr, **kwargs)
-
         self.channels = (ChNr(slot_nr),)
-
         self._measure_config: Dict[str, Optional[Any]] = {
             k: None for k in ("measure_range",)}
         self._source_config: Dict[str, Optional[Any]] = {
             k: None for k in ("output_range", "compliance",
                               "compl_polarity", "min_compliance_range")}
+        self._timing_parameters: Dict[str, Optional[Any]] = {
+            k: None for k in ("h_bias", "interval", "number", "h_base")}
 
         # We want to snapshot these configuration dictionaries
-        self._meta_attrs += ['_measure_config', '_source_config']
+        self._meta_attrs += ['_measure_config', '_source_config',
+                             '_timing_parameters']
+
+        self.add_parameter(
+            name="measurement_mode",
+            get_cmd=None,
+            set_cmd=self._set_measurement_mode,
+            set_parser=MM.Mode,
+            vals=vals.Enum(*list(MM.Mode)),
+            docstring=textwrap.dedent("""
+                Set measurement mode for this module.
+                
+                It is recommended for this parameter to use values from 
+                :class:`.constants.MM.Mode` enumeration.
+                
+                Refer to the documentation of ``MM`` command in the 
+                programming guide for more information.""")
+        )
+        # Instrument is initialized with this setting having value of
+        # `1`, spot measurement mode, hence let's set the parameter to this
+        # value since it is not possible to request this value from the
+        # instrument.
+        self.measurement_mode.raw_value = MM.Mode.SPOT
+        self.measurement_mode._save_val(MM.Mode.SPOT)
```

Comment:
Why not just pass the initial value to the constructor? It takes initial values for this reason

Reply:
passing the `initial_value` will call `set` with it on the parameter, which i decided to be undesirable some weeks ago, but now i don't remember why i decided this :( perhaps i wanted to avoid 'setting' things at "init"...

what would be your recommendation here?

Reply:
I don't like calling private methods but perhaps we just need to extend inital_value to also handle this corner case. No strong objection against leave this as is